SYCL/USM/free_during_kernel_execution.cpp

```diff
@@ -0,0 +1,41 @@
+//==------------------- free_during_kernel_execution.cpp -------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple  %s -o %t1.out
+// RUN: %HOST_RUN_PLACEHOLDER %t1.out
+// RUN: %CPU_RUN_PLACEHOLDER %t1.out
+// RUN: %GPU_RUN_PLACEHOLDER %t1.out
+// RUN: %ACC_RUN_PLACEHOLDER %t1.out
+
+#include <CL/sycl.hpp>
+
+class KernelA;
+
+int main()
+{
+    sycl::queue Queue;
+    sycl::buffer<int, 1> Buffer(64);
+    sycl::range<1> NumOfWorkItems{Buffer.size()};
+
+    auto *USM = sycl::malloc_host<int>(1, Queue.get_context());
+
+    Queue.submit([&](sycl::handler &cgh)
+    {
+        auto Accessor = Buffer.get_access<sycl::access::mode::write>(cgh);
+        cgh.parallel_for<KernelA>(NumOfWorkItems, [=](sycl::id<1> WIid)
+        {
+            Accessor[WIid] = WIid.get(0);
+        });
+    });
+
+    // Check that freeing USM before the kernel was finished works.
+    free(USM, Queue.get_context());
```

Comment:
How is the check performed?
Should one use cl or ze intercept?

Reply:
Discussed this with @s-kanaev offline, we currently don't have unit tests for OpenCL plugin in order to check what `piextUSMFree` is lowered to, so we agreed on checking that free works (i. e. does not crash) as an E2E test.